### PR TITLE
OC-1.1: Added deviation for skip telemetry check for hostname and domain-name

### PR DESCRIPTION
**Pull request by stthummuri**
Added deviation CliTakesPrecedenceOverOC in deviations.go
Added the same deviation in the script hostname_test.go to skip telemetry check for hostname and domain-name in cases where cli takes precedence over OC

### CODE REVIEW

**Review by rszarecki — APPROVED** [review]

File: feature/system/tests/system_base_test/hostname_test.go [hunk 1]
```diff
@@ -57,13 +58,16 @@ func TestHostname(t *testing.T) {
...
 			})
 
-			t.Run("Get Hostname Telemetry", func(t *testing.T) {
-				stateGot := gnmi.Await(t, dut, state.State(), 1*time.Minute, testCase.hostname)
-				if got, _ := stateGot.Val(); got != testCase.hostname {
-					t.Errorf("Telemetry hostname: got %v, want %s", stateGot, testCase.hostname)
-				}
-			})
-
+			if deviations.CliTakesPrecedenceOverOC(ondatra.DUT(t, "dut")) {
```
Comment (greg-dennis) [inline feature/system/tests/system_base_test/hostname_test.go:61]: You don't need to call ondatra.DUT(t, "dut") here -- you already have a reference to the DUT. This should just be:
`if deviations.CliTakesPrecedenceOverOC(dut)`

File: feature/system/tests/system_base_test/hostname_test.go [hunk 2]
```diff
@@ -108,13 +112,16 @@ func TestDomainName(t *testing.T) {
...
 			})
 
-			t.Run("Get Domainname Telemetry", func(t *testing.T) {
-				stateGot := gnmi.Await(t, dut, state.State(), 1*time.Minute, testCase.domainname)
-				if got, _ := stateGot.Val(); got != testCase.domainname {
-					t.Errorf("Telemetry domainname: got %v, want %s", stateGot, testCase.domainname)
-				}
-			})
-
+			if deviations.CliTakesPrecedenceOverOC(ondatra.DUT(t, "dut")) {
```
Comment (greg-dennis) [inline feature/system/tests/system_base_test/hostname_test.go:115]: `if deviations.CliTakesPrecedenceOverOC(dut)`

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: internal/deviations/deviations.go [hunk 3]
```diff
@@ -89,6 +89,11 @@ func GRIBIMACOverrideWithStaticARP(_ *ondatra.DUTDevice) bool {
 	return *gribiMACOverrideWithStaticARP
 }
```
Comment (greg-dennis) [inline internal/deviations/deviations.go:91]: "returns config" -> "returns whether config"

File: feature/system/tests/system_base_test/hostname_test.go (hunk 2 above)
Comment (stthummuri) [reply feature/system/tests/system_base_test/hostname_test.go]: Done

File: feature/system/tests/system_base_test/hostname_test.go (hunk 1 above)
Comment (stthummuri) [reply feature/system/tests/system_base_test/hostname_test.go]: Done

File: internal/deviations/deviations.go (hunk 3 above)
Comment (stthummuri) [reply internal/deviations/deviations.go:91]: Done

File: internal/deviations/deviations.go [hunk 4]
```diff
@@ -89,6 +89,11 @@ func GRIBIMACOverrideWithStaticARP(_ *ondatra.DUTDevice) bool {
...
 
+// CliTakesPrecedenceOverOC returns whether config pushed through origin CLI takes precedence over config pushed through origin OC.
+func CliTakesPrecedenceOverOC(_ *ondatra.DUTDevice) bool {
```
Comment (greg-dennis) [inline internal/deviations/deviations.go:93]: Sorry, I missed the initialism here. "CLI" should be in all caps here and in the comment right above:
https://google.github.io/styleguide/go/decisions#initialisms

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: internal/deviations/deviations.go (hunk 4 above)
Comment (stthummuri) [reply internal/deviations/deviations.go]: Done

**Review by greg-dennis — APPROVED** [review]
approving for Go quality only

**Pull request merged**